virtual-server/examples/go/main.go

```diff
@@ -0,0 +1,146 @@
+			Namespace: namespace,
+			Name:      name,
+		}, vs)
+		if err != nil {
+			log.Fatalf("cannot obtain virtual server status: %v\n", err)
+		}
+
+		cond := vs.GetReadyStatus()
+		if cond == nil {
+			return "Return"
+		} else if cond.Reason == "VirtualServerReady" && cond.Type == "Ready" && cond.Status == "True" {
+			fmt.Printf("Network:\n\tInternalIP: %s\n\tTCP: %s\n\tUDP: %s",
+				vs.Status.InternalIP(),
+				vs.Status.TCP(),
+				vs.Status.UDP(),
+			)
+			return "Ready"
+		} else if cond.Reason == "VirtualServerStopped" && cond.Type == "Ready" && cond.Status == "False" {
+			return "Stopped"
+		}
+		time.Sleep(2)
+	}
+}
+
+func main() {
+	name := "my-virtual-server"
+
+	// Get namespace or use default
+	namespace, envExist := os.LookupEnv("NAMESPACE")
+	if envExist == false {
+		namespace = "default"
+	}
+	// Uses the value of the KUBECONFIG environment variable as a filepath to a kube config file
+	c, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		log.Fatalf("Failed to create client\n")
+	}
+
+	vsv1alpha.AddToScheme(c.Scheme())
+
+	// prepare config for kubevirt client, you need to set env variable, KUBECONFIG=<path-to-kubeconfig>/.kubeconfig
+	kubevirtClientConfig := kubecli.DefaultClientConfig(&pflag.FlagSet{})
+
+	// get the kubevirt client, using which kubevirt resources can be managed.
+	kubevirtClient, err := kubecli.GetKubevirtClientFromClientConfig(kubevirtClientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain KubeVirt client: %v\n", err)
+	}
+
+	// Create a new VirtualServer with the name "my-virtual-server" to be deployed in the "default" namespace
+	virtualServer := vsv1alpha.NewVirtualServer(name, namespace)
+	// Set the region the VirtualServer will be deployed to
+	virtualServer.SetRegion("ORD1")
+	// Specify the VirtualServer operating system
+	virtualServer.SetOS(vsv1alpha.VirtualServerOSTypeLinux)
+	// Set a GPU type request for the VirtualServer
+	virtualServer.SetGPUType("Quadro_RTX_4000")
+	// Set the number of GPUs to request for the VirtualServer
+	virtualServer.SetGPUCount(1)
+	// Set the cpu core count for the VirtualServer
+	virtualServer.SetCPUCount(2)
+	//virtualServer.SetCPUType("amd-epyc-rome")
+	// Set the memory request for the VirtualServer
+	virtualServer.SetMemory("2Gi")
+	// Add a user to be added by cloudinit
+	virtualServer.AddUser(vsv1alpha.VirtualServerUser{
+		Username: "myuser",
+		Password: "password",
+	})
+	// Configure the root filesystem of the VirtualServer to clone a preexisting PVC namedubuntu1804-docker-master-20210210-ord1
+	// sourced in the vd-images namespace
+	err = virtualServer.ConfigureStorageRootWithPVCSource(vsv1alpha.VirtualServerStorageRootPVCSource{
+		Size:             "40Gi",
+		PVCName:          "ubuntu1804-docker-master-20210210-ord1",
+		PVCNamespace:     "vd-images",
+		StorageClassName: "ceph-ssd-2-replica",
+		VolumeMode:       corev1.PersistentVolumeBlock,
+		AccessMode:       corev1.ReadWriteOnce,
```

Comment:
VolumeMode Block and AccessMode ReadWriteOnce are defaults. These are therefore optional and unnecessary in this instance. We could add a comment to note that or just leave it as is. Up to you

Reply:
If they are default and I don't think we can change those 2 parameters for VM, I just removed them to not confuse users.